### PR TITLE
Add RFC 5464 response codes

### LINKED ISSUE #113: Missing ResponseCode: LONGENTRIES from METADATA extension
The METADATA IMAP extension defines a LONGENTRIES response code that should probably be added to `ResponseCode`: https://tools.ietf.org/html/rfc5464#section-4.2.1

### PR DESCRIPTION
**Pull request by jonhoo**
Fixes #113.

### CODE REVIEW

File: imap-proto/src/types.rs [hunk 1]
```diff
@@ -71,6 +71,10 @@ pub enum ResponseCode<'a> {
     AppendUid(u32, Vec<UidSetMember>),
     CopyUid(u32, Vec<UidSetMember>, Vec<UidSetMember>),
     UidNotSticky,
+    MetadataLongEntries(u64), // RFC 5464, section 4.2.1
+    MetadataMaxSize(u64),     // RFC 5464, section 4.3
+    MetadataTooMany,          // RFC 5464, section 4.3
+    MetadataNoPrivate,        // RFC 5464, section 4.3
```
Comment (djc) [inline imap-proto/src/types.rs:77]: Any thoughts on whether this ought to be a single variant that contains another enum value? Perhaps that would make it easier to handle these as a group if desired?

File: imap-proto/src/parser/rfc3501/mod.rs [hunk 2]
```diff
@@ -176,6 +176,10 @@ fn resp_text_code(i: &[u8]) -> IResult<&[u8], ResponseCode> {
             rfc4315::resp_text_code_append_uid,
             rfc4315::resp_text_code_copy_uid,
             rfc4315::resp_text_code_uid_not_sticky,
+            rfc5464::resp_text_code_metadata_longentries,
```
Comment (djc) [inline imap-proto/src/parser/rfc3501/mod.rs]: Also let's do `long_entries` etc.

File: imap-proto/src/types.rs (hunk 1 above)
Comment (jonhoo) [reply imap-proto/src/types.rs:77]: I don't _think_ they should be split out. They're not generally matched together, and if they were, it'd be _either_ the first _or_ the last three. And it seems awkward to have an enum that covers just the last three, or one that covers all four when only three of them will be matched as a group..

**Comment by djc** [comment]
Thanks!

**Comment by jonhoo** [comment]
No, thank you!

**Pull request merged**